### PR TITLE
Tooltips for messages in review panel

### PR DESCRIPTION
**Pull request by wittejm**
Do not merge before backend dependency is deployed: https://github.com/Ameelio/connect-api-nest/pull/621

https://user-images.githubusercontent.com/2104990/149038875-97d9b037-738e-4c03-91d9-a385e50ac9b6.mov

and a two-keyword example: 

two_keywords

### CODE REVIEW

File: src/components/MessageReviewPanel/MessagesList.tsx [hunk 1]
```diff
@@ -31,7 +32,11 @@ const MessagesList = ({ messages, focus }: Props) => {
             className={`p-4 ${offset} ${isFocused ? "bg-yellow-50" : ""}`}
             ref={isFocused ? focusRef : undefined}
           >
-            <MessageCard message={message} className={`max-w-lg ${offset}`} />
+            {
```
Comment (cainlevy) [inline src/components/MessageReviewPanel/MessagesList.tsx:35]: what do these braces do?

File: src/components/MessageReviewPanel/ReviewTooltip.tsx [hunk 2]
```diff
@@ -0,0 +1,80 @@
+import { Tooltip } from "antd";
+import React from "react";
+import { MessageStatus } from "src/api/graphql";
+import { formatDate } from "src/utils";
+import { TMessageItem } from "./MessageCard";
+
+interface Props {
+  message: TMessageItem;
+  children?: React.ReactNode | React.ReactNode[]
```
Comment (cainlevy) [inline src/components/MessageReviewPanel/ReviewTooltip.tsx:9]: i believe you get this for free with `React.FC<Props>`, if you like

File: src/components/MessageReviewPanel/ReviewTooltip.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+}
+
+const ReviewTooltip = ({ message, children }: Props) => {
+  const MESSAGE_AUTO_DELAY_MINUTES = 30;
+  const reviewer = message.reviewedBy ? `${message.reviewedBy?.firstName} ${message.reviewedBy?.lastName}` : "Unknown";  // This only occurs from a data error.
```
Comment (cainlevy) [inline src/components/MessageReviewPanel/ReviewTooltip.tsx:14]: the ternary should be enough:

```suggestion
  const reviewer = message.reviewedBy ? `${message.reviewedBy.firstName} ${message.reviewedBy.lastName}` : "Unknown";  // This only occurs from a data error.
```

File: src/components/MessageReviewPanel/ReviewTooltip.tsx [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    } else return null; // if skipped review; or pending review but there are no keywords (data error)
+
+  })();
+
+  const overlayInnerStyle = message.status === MessageStatus.Rejected ? { width: '400px' } : {}; // Rejected message needs more space for text.
+
+  if (tooltipContent) {
+    return (
+      <Tooltip title={tooltipContent} trigger="click" overlayInnerStyle={overlayInnerStyle}>
+        {" " /* The tooltip doesn't show up unless I... add this invisible space. what the heck?*/}
```
Comment (cainlevy) [inline src/components/MessageReviewPanel/ReviewTooltip.tsx:70]: 👀 fun

**Review by cainlevy — APPROVED** [review]

File: src/components/MessageReviewPanel/MessagesList.tsx (hunk 1 above)
Comment (wittejm) [reply src/components/MessageReviewPanel/MessagesList.tsx:35]: .Ah! Nothing. an artifact of a different approach

File: src/components/MessageReviewPanel/ReviewTooltip.tsx (hunk 2 above)
Comment (wittejm) [reply src/components/MessageReviewPanel/ReviewTooltip.tsx:9]: That's what I've been led to believe, but I haven't gotten it to work that way and I haven't found working examples of that in our code :shrug:

Comment (cainlevy) [reply src/components/MessageReviewPanel/ReviewTooltip.tsx:9]: huh. what does not working look like? i've never had a problem with it (in previous projects).

**Pull request merged**